### PR TITLE
Add i18n page

### PR DESCRIPTION
**Pull request by benhalpern**
This documentation is meant to accompany changes made in this PR:

https://github.com/forem/forem/pull/14620

And act as guidance for further development of this feature. This documentation should change as this functionality stabilizes as there are several references in the docs to best practices not yet entirely established.

### CODE REVIEW

**Review by cmgorton — APPROVED** [review]

File: docs/technical-overview/i18n.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+```html
+<div><%= t("core.add_comment") %></div>
+```
+
+`t` in this view helper context is short for `I18n.translate`. Check out the linked Rails guide for full information about how to use all standard translation and localization functionality.
+
+### i18n.js
+
+We use the [i18n.js](https://github.com/fnando/i18n-js) gem to pass values to JavaScript. Our frontend code is currently divided into old asset pipeline code in `app/assets/javascripts` and modern Rails JS practices at `app/javascript`. In the asset pipeline you should directly use `I18n.t` and in app/javascript, there is a wrapper at `app/javascript/utilities/locale.js` you can include as you go. This is all subject to change in the name of code quality improvements.
```
Comment (juliannatetreault) [inline docs/technical-overview/i18n.md:49]: _**Quick, non-blocking suggestion:**_

```suggestion
We use the [i18n.js](https://github.com/fnando/i18n-js) gem to pass values to JavaScript. Our frontend code is currently divided into old asset pipeline code in `app/assets/javascripts` and modern Rails JS practices at `app/javascript`. In the asset pipeline you should directly use `I18n.t` and in `app/javascript`, there is a wrapper at `app/javascript/utilities/locale.js` you can include as you go. This is all subject to change in the name of code quality improvements.
```

File: docs/technical-overview/i18n.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+This should generally be abstracted away for you if you follow above patterns, but for your information, we store translatiions in the view like this:
```
Comment (juliannatetreault) [inline docs/technical-overview/i18n.md:51]: _Quick, typo fix:_

```suggestion
This should generally be abstracted away for you if you follow above patterns, but for your information, we store translations in the view like this:
```

File: docs/technical-overview/i18n.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+sidebar_position: 7
+---
+
+# Internationalization (i18n)
+
+Forem's internationalization is currently in its early stages. You are *encouraged* to follow patterns laid out here, but it is not yet *required* for every pull request until further development and stabilization of the feature.
+
+This development is a matter of translation as well as de-coupling of copy from code. As this functionality is new, best practices in this area are still a work in progress.
+
+## I18n features and direction
```
Comment (juliannatetreault) [inline docs/technical-overview/i18n.md:11]: _Quick, non-blocking question:_ should the "i" in "i18n" here be lowercase?
```suggestion
## i18n features and direction
```

**Review by juliannatetreault — APPROVED** [review]
Left a few suggestions, otherwise, looks great to me!

File: docs/technical-overview/i18n.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
+---
+sidebar_position: 7
+---
+
+# Internationalization (i18n)
+
+Forem's internationalization is currently in its early stages. You are *encouraged* to follow patterns laid out here, but it is not yet *required* for every pull request until further development and stabilization of the feature.
+
+This development is a matter of translation as well as de-coupling of copy from code. As this functionality is new, best practices in this area are still a work in progress.
```
Comment (juliannatetreault) [inline docs/technical-overview/i18n.md:9]: _One more quick suggestion!:_

```suggestion
This development is a matter of translation as well as decoupling of copy from code. As this functionality is new, best practices in this area are still a work in progress.
```

**Pull request merged**